### PR TITLE
Add TLS support

### PR DESCRIPTION
**Pull request by maelk**
This commit allows the user to start Inspector using TLS on all
endpoints, by setting the following environment variables:

- CACERT_FILE: path to the CA cert
- CERT_FILE: path to the cert
- KEY_FILE: path to the key

### CODE REVIEW

**Review by dtantsur — CHANGES_REQUESTED** [review]
Same comments as on ironic-image.

**Comment by maelk** [comment]
I'll address them similarly, thank you.

File: runironic-inspector.sh [hunk 1]
```diff
@@ -1,15 +1,41 @@
...
-crudini --set $CONFIG ironic endpoint_override http://$IRONIC_URL_HOST:6385
-crudini --set $CONFIG service_catalog endpoint_override http://$IRONIC_URL_HOST:5050
+if [ ! -z "$CERT_FILE" ] || [ ! -z "$KEY_FILE" ] || [ ! -z "$CACERT_FILE" ] || [ ! -z "$INSECURE" ]; then
+    crudini --merge $CONFIG <<EOF
+[DEFAULT]
+host = $IRONIC_IP
+[ironic]
+endpoint_override = https://${IRONIC_URL_HOST}:6385
+$([ ! -z "$CACERT_FILE" ] && echo "cafile = $CACERT_FILE")
+$([ ! -z "$INSECURE" ] && echo "insecure = $INSECURE" || echo "insecure = false")
```
Comment (kashifest) [inline runironic-inspector.sh]: even when INSECURE is false the value is set right? is this condition correct?

Comment (maelk) [reply runironic-inspector.sh]: it's rather "if insecure is set then set it to the value, otherwise set it to false"

**Review by dtantsur — CHANGES_REQUESTED** [review]
Same comment as on ironic: let's use jinja and default CACERT to CERT.

**Comment by maelk** [comment]
/test-integration

@dtantsur I'm looking into putting inspector behind Apache, we have two options here, simple reverse proxy or using WSGI. I prefer the second one because we don't need to have inspector listening on the loopback interface, but it requires additional complexity for inspector, to separate the api element. What's your opinion? I'm leaning towards the simple reverse proxy.

**Comment by dtantsur** [comment]
> I'm leaning towards the simple reverse proxy.

Me too, at least for the time being.

File: runhealthcheck.sh [hunk 2]
```diff
@@ -1,5 +1,8 @@
 #!/bin/bash
 set -ex
 
-
-curl -s http://127.0.0.1:5050
\ No newline at end of file
+if [ -z "$CERT_FILE" ]; then
+    curl -s http://127.0.0.1:5050
+else
+    curl -k -s https://127.0.0.1:5050
```
Comment (dtantsur) [inline runhealthcheck.sh]: You could use `--cacert "$CERT_FILE"`

**Review by dtantsur — APPROVED** [review]

File: runhealthcheck.sh (hunk 2 above)
Comment (maelk) [reply runhealthcheck.sh]: good point, I'll change it. thanks

**Comment by maelk** [comment]
@zaneb @dhellmann @derekhiggins Any input on this one ? lgty ? thanks

File: runironic-inspector.sh [hunk 3]
```diff
@@ -1,16 +1,34 @@
...
 
+if [ -n "$CERT_FILE" ]; then
+    export IRONIC_BASE_URL="https://${IRONIC_URL_HOST}"
+    export CACERT_FILE="${CACERT_FILE:-$CERT_FILE}"
+else
+    export IRONIC_BASE_URL="http://${IRONIC_URL_HOST}"
+fi
+
+
 cp $CONFIG $CONFIG.orig
```
Comment (derekhiggins) [inline runironic-inspector.sh:47]: the value for $CONFIG is being change above, will this cp still work?

**Comment by maelk** [comment]
@dtantsur, I have now separated the variables for ironic and inspector, so that we can give different certificates, similarly to what is done with basic authentication where we can configure it independently for ironic and inspector. I did the same modification on https://github.com/metal3-io/ironic-image/pull/198

File: runironic-inspector.sh (hunk 3 above)
Comment (maelk) [reply runironic-inspector.sh:47]: yes. The change above is just to switch to use the expected name for the config file by Oslo.

**Comment by maelk** [comment]
/test-integration
@derekhiggins I have added the same fix here.
adding @dtantsur @dhellmann @hardys , How does this look to you ?

**Comment by maelk** [comment]
CI glitch, re-triggering
/test-integration

**Comment by maelk** [comment]
@derekhiggins @dtantsur @zaneb Could you please have a look at this PR ? I would like to proceed with the TLS string of PRs.

**Review by dtantsur — APPROVED** [review]
LGTM, not approving to give @derekhiggins a chance to take another look.

**Comment by derekhiggins** [comment]
> LGTM, not approving to give @derekhiggins a chance to take another look.

lgtm also, there is a small fix needed on the ironic-image PR so held off on merging until both can be done together.

**Pull request merged**